vital/types/point.h

```diff
@@ -38,36 +39,48 @@
 #ifndef KWIVER_VITAL_POINT_H_
 #define KWIVER_VITAL_POINT_H_
 
-#include <vital/vital_config.h>
-#include <vital/vital_export.h>
 #include <vital/types/covariance.h>
 #include <vital/types/vector.h>
 
+#include <vital/vital_config.h>
+#include <vital/vital_export.h>
+
 #include <memory>
 
 namespace kwiver {
+
 namespace vital {
 
 template < unsigned N, typename T >
 class VITAL_EXPORT point
 {
 public:
   using vector_type = Eigen::Matrix< T, N, 1 >;
-  using covariance_type = covariance_<N, float>;
+  using covariance_type = covariance_< N, float >;
+
+  point() {}
+  point( vector_type const& v, covariance_type const& c = {} )
```

Comment:
should this be `explicit` to prevent implicit casting from vectors to points?

Reply:
Should it? The point of this PR is to make it easier to create a `point`, e.g. by making this work: `point_3d{ x, y, z }`. Basically, it is making it easier to construct a `point` when you don't care about the covariance. It would seem in line with that to allow passing a `vector_Xd` to a function that wants a `point_Xd`.

Reply:
There are semantic differences between a "point" and a "vector" even though you can represent them both as an array of numerical values.  They have different operations.  You can add two vectors, but you can not add two points.  You can subtract one point from another and the result is a vector.  You can add a add a vector to a point and the result is a point, etc.  Not that we support all of these operations yet, but we might at some point.  VXL (vgl) has separate notions of point and vector that enforce these properties and does not allow implicit casting between the two.  I like that approach, but I may be a geometry purist.

Reply:
Hmm, I see, but unfortunately AFAIK `vector` is already overloaded that way. The difference between `kwiver::point_Xd` and `kwiver::vector_Xd` is not the difference between "point" and "vector" in the geometry sense, but that one has covariance and the other does not. That is, users that need a point, but do not need/want covariance, already use `vector_Xd`.

(This reminds me of GRAVL's timestamp class, which explicitly had a `time_point` and `time_delta`, and operator overloads enforced not mixing them improperly.)

Reply:
Yes, we have used `vector` in the past in a loose sense as both a location and a change in location.  However, if we are introducing a class called `point` now with the specific purpose of representing point locations (and uncertainty), then I will argue that we should start being more explicit about the different between a point and a vector.

Reply:
I would agree that we should provide/enforce some firm semantics in this area.